src/Compiler/Facilities/prim-lexing.fs

```diff
@@ -108,6 +110,24 @@ type StringText(str: string) =
         member _.CopyTo(sourceIndex, destination, destinationIndex, count) =
             str.CopyTo(sourceIndex, destination, destinationIndex, count)
 
+        member this.GetSubTextFromRange(range) =
```

Comment:
What happens if invalid range is passed? Or zero range?

Reply:
Hmm, we throw an invalid argument exception?
Return empty string for zero range?

Reply:
Good question. We can return ValueOption for example, or empty string (latter would have the same behaviour as "zero" range, probably).

Reply:
I'm not in favour of being very defensive and returning `ValueOption`.
It would be inconsistent compared to 
https://github.com/dotnet/fsharp/blob/048524ccffe1421a404f0af0ac5540f9a7353bc1/src/Compiler/Facilities/prim-lexing.fs#L86
You can pass in any integer there right now, which also isn't validated there whatsoever.

I've used this extension method for years now and have never even passed in any invalid range.
I don't want to be bothered with dealing with options because some users might pass in invalid input. The typical use case for this scenario is getting the text via a range you found in the AST anyway.